scripts/dotnet.py

```diff
@@ -625,18 +627,22 @@ def get_commit_date(
         url = urlformat % (owner, repo, commit_sha)
 
     build_timestamp = None
-    sleep_time = 10 # Start with 10 second sleep timer
-    for retrycount in range(5):
+    for retrycount in range(2): # Only retry once
         try:
             with urlopen(url) as response:
                 getLogger().info("Commit: %s", url)
                 item = loads(response.read().decode('utf-8'))
                 build_timestamp = item['commit']['committer']['date']
                 break
         except URLError as error:
-            getLogger().warning(f"URL Error trying to get commit date from {url}; Reason: {error.reason}; Attempt {retrycount}")
-            sleep(sleep_time)
-            sleep_time = sleep_time * 2
+            with urlopen(ratelimiturl) as response:
+                item = loads(response.read().decode('utf-8'))
+                reset_timestamp = item['resources']['core']['reset']
+                remaining_calls = item['resources']['core']['remaining']
+                seconds_to_retry = reset_timestamp - time.time() + 10 # Add 10 seconds to be safe
+                getLogger().warning(f"URL Error trying to get commit date from {url}; Reason: {error.reason}; Attempt {retrycount}; reset time: {reset_timestamp} (Unix Time); time to wait: {seconds_to_retry} seconds; remaining calls: {remaining_calls}")
+                if(remaining_calls == 0 and retrycount == 0):
```

Comment:
So this seems to only wait if `remaining_calls` is equal to zero. I assume that means that we have hit the retry limit from the GitHub API. Are we sure that we don't want to have any kind of sleep in the case where we are not at the limit? Or am I just missing something here?

Reply:
We should probably add a sleep of some sort, I will add one for 30 seconds. I think it could go either way with adding a sleep as if we still have remaining calls, something else likely failed, instead of having exceeded the API rate limit.

Reply:
Yeah, I think that we really want this retry and backoff for two things. One is to deal with transient network failures. And in that case we just want to wait some set amount of time and try again. However, if we are being rate limited then we will need to use the more complex call that we have in this PR. I wonder if it would be worthwhile to break it out into those two paths?